ui/components/component-library/button-icon/button-icon.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Box from '../../ui/box';
+import { Icon } from '../icon';
+
+import {
+  ALIGN_ITEMS,
+  DISPLAY,
+  JUSTIFY_CONTENT,
+} from '../../../helpers/constants/design-system';
+import { BUTTON_ICON_SIZES } from './button-icon.constants';
+
+export const ButtonIcon = ({
+  as = 'button',
+  className,
+  size = BUTTON_ICON_SIZES.LG,
+  icon,
+  disabled,
+  primary,
+  iconProps,
+  ...props
+}) => {
+  return (
+    <Box
+      as={as}
+      padding={1}
+      className={classnames(
+        'mm-button-icon',
+        `mm-button-icon--size-${size}`,
+        {
+          'mm-button-icon--disabled': disabled,
+          'mm-button-icon--type-primary': primary,
+        },
+        className,
+      )}
+      disabled={disabled}
+      display={DISPLAY.INLINE_FLEX}
+      justifyContent={JUSTIFY_CONTENT.CENTER}
+      alignItems={ALIGN_ITEMS.CENTER}
+      {...props}
+    >
+      <Icon name={icon} size={size} {...iconProps} />
+    </Box>
+  );
+};
+
+ButtonIcon.propTypes = {
+  /**
+   * The polymorphic `as` prop allows you to change the root HTML element of the Button component between `button` and `a` tag
+   */
+  as: PropTypes.string,
+  /**
+   * An additional className to apply to the ButtonIcon.
+   */
+  className: PropTypes.string,
+  /**
+   * Boolean to disable button
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The name of the icon to display. Should be one of ICON_NAMES
+   */
+  icon: PropTypes.string, // Can't set PropTypes.oneOf(ICON_NAMES) because ICON_NAMES is an environment variable
```

Comment:
Correct you cannot 😢 but what happens if we don't provide an icon name? Should this be a required prop? It is for the icon component

```suggestion
  icon: PropTypes.string.isRequired, // Can't set PropTypes.oneOf(ICON_NAMES) because ICON_NAMES is an environment variable
```